### PR TITLE
Docs/Guide for running node in production

### PR DESCRIPTION
**Pull request by hanjukim**
## Summary of changes

* Added Minimum Hard Requirements section in `Installation` page.
* Added `Running node in production` page.

## Report of required housekeeping

- [ ] Github issue OR spec proposal link
- [ ] Wrote tests
- [x] Updated relevant documentation (docs/)
- [ ] Added a relevant changelog entry: clog add [section] [stanza] [message]

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

File: docs/guide/README.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ This section will guide the reader to install, deploy, and interact with the Ter
 
 * [**Installation**](installation.md): Instructions on how to install _terrad_, the server daemon for nodes, and _terracli_, the cli tool to interact with terrad. 
 * [**How to Join the network**](join-network.md): Columbus is the first official public network running the Terra Protocol. This section outlines step by step instructions on how to start operating a node on Columbus. 
+* [**Running node in production**](production.md): Guides on running node in production.
```
Comment (etienne-napoleone) [inline docs/guide/README.md]:
```suggestion
* [**Running node in production**](production.md): Guides on running a node in production.
```

File: docs/guide/installation.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 This guide will explain how to install the `terrad` and `terracli` entrypoints onto your system. With these installed on a server, you can participate in the latest testnet as either a [Full Node](join-network.md#run-a-full-node) or a [Validator](https://github.com/terra-project/core/tree/f8be66ca87e5a7d50a28875f1bea04dbfe69b9c6/docs/guide/setup-validator.md).
 
+## Minimum Hardware Requirements
+Hardware requirements for running a node:
+
+* CPU cores: 2 or more
+* Storage: 128G or more
+* Network Bandwidth: 2.5 ~ 5Mbps (more traffic can be used at syncing up)
```
Comment (etienne-napoleone) [inline docs/guide/installation.md]:
```suggestion
* Network Bandwidth: 2.5 ~ 5Mbps (more traffic can be used while syncing up)
```

File: docs/guide/production.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Running for production
+In this section, we will talk about how to run a node for production. At the moment we will focus on Linux (Redhat Compatiable) only.
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
In this section, we will talk about how to run a node for production. At the moment we will focus on Linux (RPM based) only.
```

File: docs/guide/production.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+`terrad` does not require super user account. We **strongly** recommend to run `terrad` as a normal user. However, at the configuration, we need super user permission for creating/modifying files.
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
`terrad` does not require super user account. We **strongly** recommend to run `terrad` as a normal user. However, during the setup process, we need super user permission for creating/modifying files.
```

File: docs/guide/production.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+This guide is for general purpose only. We recommend to read [Validator](https://docs.terra.money/features/overview) section for operating validator node.
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
This guide is for general purpose only. We recommend to read [Validator](https://docs.terra.money/features/overview) section for operating a validator node.
```

File: docs/guide/production.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+## Firewall configuration
+`terrad` uses several TCP port for each purposes.
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
`terrad` uses several TCP ports for each purposes.
```

File: docs/guide/production.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+* `26656` is default port for P2P protocol. This port has to be opened to communicate with other nodes. This port must be opened for joining the network. **However,** it does not have to be opened to public. For validator nodes, we recommened to configure `persistent_peers` and close this port to public.
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
* `26656` is the default port for P2P protocol. This port has to be opened to communicate with other nodes. This port must be opened for joining the network. **However,** it does not have to be opened to the public. For validator nodes, we recommend to configure `persistent_peers` and close this port to the public.
```
This part is a bit confusing for me.
Is this port used only for peers discovery?
In this sentence it sounds like it's used for communicating data with other nodes.

File: docs/guide/production.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+* `26657` is default port for RPC protocol. This port is used for querying / sending transaction. In other words, this port needs to be opened for serving query from `terracli`. It is safe _NOT_ to open this port to the public unless you are planning to run public node.
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
* `26657` is the default port for RPC protocol. This port is used for querying / sending transactions. In other words, this port needs to be opened for serving queries from `terracli`. It is safe _NOT_ to open this port to the public unless you are planning to run public node.
```

File: docs/guide/production.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+* `1317` is default port for Lite Client Daemon(LCD), whichcan be executed by `terracli rest-server`. LCD provides HTTP RESTFul API layer to interact with `terrad` node(RPC). You can check `https://lcd.terra.dev/swagger-ui/` out for example. Again, you don't need to open this port unless you need to.
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
* `1317` is the default port for Lite Client Daemon(LCD), which can be executed by `terracli rest-server`. LCD provides HTTP RESTful API layer to interact with `terrad` node(RPC). You can check `https://lcd.terra.dev/swagger-ui/` out for examples. Again, you don't need to open this port unless you have use of it.
```

File: docs/guide/production.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+* `26660` is default port for interacting with [Prometheus](https://prometheus.io) database which can be used for monitoring environment. This port is not opened in default configuration.
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
* `26660` is the default port for interacting with the [Prometheus](https://prometheus.io) database which can be used for monitoring the environment. This port is not opened in the default configuration.
```

File: docs/guide/production.md [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+## Unlimit the maximum number of opened files for one process
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
## Raise the maximum number of opened files for one process
```

File: docs/guide/production.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+`terrad` can open more than 1024(which is default max opened file configuration) files at some moment. We should increase this limit. Modify `/etc/security/limits.conf` to unlimit `nofile`
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
`terrad` can open more than 1024 files (which is default maximum) concurrently.
We should increase this limit.
Modify `/etc/security/limits.conf` to raise the `nofile` capability.
```

File: docs/guide/production.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+ExecStart=/data/terra/go/bin/terrad start
+Restart=on-abort
+
+[Install]
+WantedBy=multi-user.target
+
+[Service]
+LimitNOFILE=65535
+```
+Modify `Service` Section from above given sample to suit your settings. Note that even we unlimited # of open files for a process, we still need `LimitNOFILE` section.
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
Modify the `Service` Section from the above given sample to suit your settings.
Note that even if we raised the number of open files for a process, we still need the `LimitNOFILE` section.
```

File: docs/guide/production.md [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+Use systemctl to control (start, stop, restart)
+
+* Start: `systemctl start terrad`
+* Stop: `systemctl stop terrad`
+* Restart: `systemctl restart terrad`
+
+#### Accessing log file
+* Entire log: `journalctl -t terrad`
+* Entire log reversed: `journalctl -t terrad -r`
+* Latest and continuous: `journalctl -t terrad -f
```
Comment (etienne-napoleone) [inline docs/guide/production.md]:
```suggestion
* Latest and continuous: `journalctl -t terrad -f`
```

**Comment by hanjukim** [comment]
@etienne-napoleone yup. we should

**Pull request merged**